commit 598461c1395904f1e7a10414b6ad94fddaf02521
Author:     Dan Davison <dandavison7@gmail.com>
AuthorDate: Wed Jun 26 11:07:15 2019 -0400
Commit:     Dan Davison <dandavison7@gmail.com>
CommitDate: Wed Jun 26 21:19:12 2019 -0400

    CLI

diff --git a/Cargo.toml b/Cargo.toml
index 91dd2c4..8d049d2 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,4 +7,5 @@ edition = "2018"
 [dependencies]
 console = "0.7.7"
 parsepatch = "0.2.0"
+structopt = "0.2.16"
 syntect = "3.2"
diff --git a/src/main.rs b/src/main.rs
index 4ad7569..2af0f85 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,8 +1,11 @@
+extern crate structopt;
+
 use std::io::{self, BufRead, ErrorKind};
 use std::path::Path;
 use std::process;
 
 use console::strip_ansi_codes;
+use structopt::StructOpt;
 use syntect::easy::HighlightLines;
 use syntect::highlighting::{Color, Style, ThemeSet};
 use syntect::parsing::{SyntaxReference, SyntaxSet};
@@ -23,6 +26,26 @@ const RED: Color = Color {
     a: 0x00,
 };
 
+#[derive(StructOpt, Debug)]
+#[structopt(name = "delta")]
+struct Opt {
+    /// Use diff highlighting colors appropriate for a light terminal
+    /// background
+    #[structopt(long = "light")]
+    light: bool,
+
+    /// Use diff highlighting colors appropriate for a dark terminal
+    /// background
+    #[structopt(long = "dark")]
+    dark: bool,
+
+    /// The width (in characters) of the diff highlighting. By
+    /// default, the highlighting extends to the last character on
+    /// each line
+    #[structopt(short = "-w", long = "width")]
+    width: Option<u16>,
+}
+
 #[derive(PartialEq)]
 enum State {
     Commit,
@@ -54,6 +77,7 @@ fn delta() -> std::io::Result<()> {
     let mut did_emit_line: bool;
     let stdin = io::stdin();
     let mut stdout = io::stdout();
+    let opt = Opt::from_args();
 
     for _line in stdin.lock().lines() {
         let raw_line = _line?;
