commit 673c73532af9dff46ae4193e02e1b21fad294732
Author:     Dan Davison <dandavison7@gmail.com>
AuthorDate: Sun Jun 23 20:10:53 2019 -0400
Commit:     Dan Davison <dandavison7@gmail.com>
CommitDate: Sun Jun 23 21:29:47 2019 -0400

    syntax-highlight hunks

diff --git a/Cargo.toml b/Cargo.toml
index 31218b1..2f10b66 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -5,4 +5,5 @@ authors = ["Dan Davison <dandavison7@gmail.com>"]
 edition = "2018"
 
 [dependencies]
+syntect = "3.2"
 unidiff = "0.3.1"
\ No newline at end of file
diff --git a/src/main.rs b/src/main.rs
index 813760f..40d32a5 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,18 +1,33 @@
 extern crate unidiff;
 
 use std::io::{self, Read, Write};
+use std::path::Path;
 
+use syntect::easy::HighlightLines;
+use syntect::highlighting::{ThemeSet, Style};
+use syntect::parsing::SyntaxSet;
+use syntect::util::{as_24_bit_terminal_escaped, LinesWithEndings};
 use unidiff::PatchSet;
 
 fn main() {
+    let ps = SyntaxSet::load_defaults_newlines();
+    let ts = ThemeSet::load_defaults();
+
     let mut input = String::new();
     io::stdin().read_to_string(&mut input);
     let mut patch_set = PatchSet::new();
     patch_set.parse(&mut input).ok().expect("Error parsing diff");
     for patched_file in patch_set {
+        let path = Path::new(&patched_file.source_file);
+        let syntax = ps.find_syntax_by_extension(path.extension().unwrap().to_str().unwrap()).unwrap();
+        let mut h = HighlightLines::new(syntax, &ts.themes["base16-ocean.dark"]);
         io::stdout().write_all(format!("\n<<<<<<<<<<<<<<<<<<<<<< file: {}\n", patched_file.source_file).as_bytes());
         for hunk in patched_file {
-            io::stdout().write_all(hunk.to_string().as_bytes());
+            for line in LinesWithEndings::from(&hunk.to_string()) {
+                let ranges: Vec<(Style, &str)> = h.highlight(line, &ps);
+                let escaped = as_24_bit_terminal_escaped(&ranges[..], true);
+                println!("{}", escaped);
+            }
         }
     }
 }
