commit 0f90659f67558cf05478fc7573814ada46ea13b0
Author:     Dan Davison <dandavison7@gmail.com>
AuthorDate: Tue Jun 25 08:37:33 2019 -0400
Commit:     Dan Davison <dandavison7@gmail.com>
CommitDate: Tue Jun 25 08:49:18 2019 -0400

    rustfmt

diff --git a/src/main.rs b/src/main.rs
index cb4335d..b81b752 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -7,10 +7,10 @@ use std::path::Path;
 use syntect::easy::HighlightLines;
 use syntect::highlighting::{Color, Style, ThemeSet};
 use syntect::parsing::SyntaxSet;
-use syntect::util::{LinesWithEndings};
+use syntect::util::LinesWithEndings;
 use unidiff::PatchSet;
 
-pub const DELTA_THEME_DEFAULT: &str = "InspiredGitHub";  // base16-mocha.dark
+pub const DELTA_THEME_DEFAULT: &str = "InspiredGitHub"; // base16-mocha.dark
 
 const GREEN: Color = Color {
     r: 0xd0,
@@ -30,16 +30,22 @@ fn main() {
     let ps = SyntaxSet::load_defaults_newlines();
     let ts = ThemeSet::load_defaults();
     let theme = &ts.themes[DELTA_THEME_DEFAULT];
-
     let mut input = String::new();
-    io::stdin().read_to_string(&mut input).expect("Error reading input");
+    io::stdin().read_to_string(&mut input).expect(
+        "Error reading input",
+    );
     let mut patch_set = PatchSet::new();
-    patch_set.parse(&mut input).ok().expect("Error parsing input as a diff");
+    patch_set.parse(&mut input).ok().expect(
+        "Error parsing input as a diff",
+    );
     for patched_file in patch_set {
         // TODO: use both source and target to determine language
         let path = Path::new(&patched_file.target_file);
-        let extension = path.extension()
-            .expect(format!("Error determining file type: {}", path.to_str().unwrap()).as_str());
+        let extension =
+            path.extension().expect(
+                format!("Error determining file type: {}", path.to_str().unwrap())
+                    .as_str(),
+            );
         let extension_str = extension.to_str().unwrap();
         let syntax = ps.find_syntax_by_extension(extension_str);
 
@@ -58,7 +64,7 @@ fn main() {
                         print!("{}", escaped);
                     }
                 }
-            },
+            }
             None => {
                 for hunk in patched_file {
                     for line in LinesWithEndings::from(&hunk.to_string()) {
@@ -75,20 +81,25 @@ fn my_as_24_bit_terminal_escaped(v: &[(Style, &str)], background_color: Option<C
     let mut s: String = String::new();
     for &(ref style, text) in v.iter() {
         match background_color {
-            Some(background_color) => write!(s,
-                                             "\x1b[48;2;{};{};{}m",
-                                             background_color.r,
-                                             background_color.g,
-                                             background_color.b).unwrap(),
+            Some(background_color) => {
+                write!(
+                    s,
+                    "\x1b[48;2;{};{};{}m",
+                    background_color.r,
+                    background_color.g,
+                    background_color.b
+                ).unwrap()
+            }
             None => (),
         }
-        write!(s,
-               "\x1b[38;2;{};{};{}m{}",
-               style.foreground.r,
-               style.foreground.g,
-               style.foreground.b,
-               text)
-            .unwrap();
+        write!(
+            s,
+            "\x1b[38;2;{};{};{}m{}",
+            style.foreground.r,
+            style.foreground.g,
+            style.foreground.b,
+            text
+        ).unwrap();
     }
     s.push_str("\x1b[0m");
     s
