commit 3692e62eb83278e60cf257445589622d7155ec35
Author:     Dan Davison <dandavison7@gmail.com>
AuthorDate: Thu Jun 27 10:54:46 2019 -0400
Commit:     Dan Davison <dandavison7@gmail.com>
CommitDate: Thu Jun 27 11:11:46 2019 -0400

    Tweak default colors

diff --git a/src/paint.rs b/src/paint.rs
index 8070db1..542207d 100644
--- a/src/paint.rs
+++ b/src/paint.rs
@@ -27,13 +27,13 @@ const LIGHT_THEME_MINUS_COLOR: Color = Color {
 
 const DARK_THEME_PLUS_COLOR: Color = Color {
     r: 0x01,
-    g: 0x18,
-    b: 0x00,
+    g: 0x3B,
+    b: 0x01,
     a: 0x00,
 };
 
 const DARK_THEME_MINUS_COLOR: Color = Color {
-    r: 0x24,
+    r: 0x3F,
     g: 0x00,
     b: 0x01,
     a: 0x00,
