commit ebdd13c9cba21f0eb4f2006962516f48d532ff8c
Author:     Dan Davison <dandavison7@gmail.com>
AuthorDate: Thu Jun 27 10:45:17 2019 -0400
Commit:     Dan Davison <dandavison7@gmail.com>
CommitDate: Thu Jun 27 11:11:46 2019 -0400

    Parse color arguments

diff --git a/src/main.rs b/src/main.rs
index 3274ced..bdf9507 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -5,10 +5,11 @@ mod parse_diff;
 
 use std::io::{self, BufRead, ErrorKind};
 use std::process;
+use std::str::FromStr;
 
 use console::strip_ansi_codes;
 use structopt::StructOpt;
-use syntect::highlighting::ThemeSet;
+use syntect::highlighting::{Color, ThemeSet};
 use syntect::parsing::{SyntaxReference, SyntaxSet};
 
 #[derive(StructOpt, Debug)]
@@ -113,6 +114,12 @@ fn delta() -> std::io::Result<()> {
             }
         }
     }
+    let minus_color = opt.minus_color.as_ref().and_then(
+        |s| Color::from_str(s).ok(),
+    );
+    let plus_color = opt.plus_color.as_ref().and_then(
+        |s| Color::from_str(s).ok(),
+    );
     let theme_name = opt.theme.unwrap();
     let theme = &theme_set.themes[&theme_name];
 
@@ -140,8 +147,8 @@ fn delta() -> std::io::Result<()> {
                         &syntax_set,
                         theme,
                         &theme_name,
-                        &opt.minus_color,
-                        &opt.plus_color,
+                        plus_color,
+                        minus_color,
                         &mut output,
                     );
                     writeln!(stdout, "{}", output)?;
diff --git a/src/paint.rs b/src/paint.rs
index af32086..8070db1 100644
--- a/src/paint.rs
+++ b/src/paint.rs
@@ -11,28 +11,28 @@ pub const DARK_THEMES: [&str; 4] = [
     "base16-ocean.dark",
 ];
 
-const LIGHT_THEME_GREEN: Color = Color {
+const LIGHT_THEME_PLUS_COLOR: Color = Color {
     r: 0xd0,
     g: 0xff,
     b: 0xd0,
     a: 0x00,
 };
 
-const LIGHT_THEME_RED: Color = Color {
+const LIGHT_THEME_MINUS_COLOR: Color = Color {
     r: 0xff,
     g: 0xd0,
     b: 0xd0,
     a: 0x00,
 };
 
-const DARK_THEME_GREEN: Color = Color {
+const DARK_THEME_PLUS_COLOR: Color = Color {
     r: 0x01,
     g: 0x18,
     b: 0x00,
     a: 0x00,
 };
 
-const DARK_THEME_RED: Color = Color {
+const DARK_THEME_MINUS_COLOR: Color = Color {
     r: 0x24,
     g: 0x00,
     b: 0x01,
@@ -46,18 +46,18 @@ pub fn paint_line(
     syntax_set: &SyntaxSet,
     theme: &Theme,
     theme_name: &String,
-    minus_color: &Option<String>,
-    plus_color: &Option<String>,
+    plus_color: Option<Color>,
+    minus_color: Option<Color>,
     buf: &mut String,
 ) {
     let mut highlighter = HighlightLines::new(syntax, theme);
     let first_char = line.chars().next();
     let is_dark = DARK_THEMES.contains(&theme_name.as_str());
     let background_color = match (first_char, is_dark) {
-        (Some('+'), true) => Some(DARK_THEME_GREEN),
-        (Some('-'), true) => Some(DARK_THEME_RED),
-        (Some('+'), false) => Some(LIGHT_THEME_GREEN),
-        (Some('-'), false) => Some(LIGHT_THEME_RED),
+        (Some('+'), true) => plus_color.or_else(|| Some(DARK_THEME_PLUS_COLOR)),
+        (Some('-'), true) => minus_color.or_else(|| Some(DARK_THEME_MINUS_COLOR)),
+        (Some('+'), false) => plus_color.or_else(|| Some(LIGHT_THEME_PLUS_COLOR)),
+        (Some('-'), false) => minus_color.or_else(|| Some(LIGHT_THEME_MINUS_COLOR)),
         _ => None,
     };
     if first_char == Some('+') || first_char == Some('-') {
