commit 8dd83d321cb803f7180549d82dcff89ed7710814
Author:     Dan Davison <dandavison7@gmail.com>
AuthorDate: Sun Jun 30 23:02:52 2019 -0400
Commit:     Dan Davison <dandavison7@gmail.com>
CommitDate: Sun Jun 30 23:02:52 2019 -0400

    Rearrange

diff --git a/src/main.rs b/src/main.rs
index a2a7ac3..311076b 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -114,52 +114,6 @@ fn main() -> std::io::Result<()> {
     Ok(())
 }
 
-fn delta(
-    lines: impl Iterator<Item = String>,
-    paint_config: &paint::Config,
-    assets: &HighlightingAssets,
-) -> std::io::Result<()> {
-
-    let mut syntax: Option<&SyntaxReference> = None;
-    let mut output = String::new();
-    let mut output_type =
-        OutputType::from_mode(PagingMode::QuitIfOneScreen, Some(paint_config.pager)).unwrap();
-    let writer = output_type.handle().unwrap();
-    let mut state = State::Unknown;
-    let mut did_emit_line: bool;
-
-    for raw_line in lines {
-        let line = strip_ansi_codes(&raw_line).to_string();
-        did_emit_line = false;
-        if line.starts_with("diff --") {
-            state = State::DiffMeta;
-            syntax = match parse_diff::get_file_extension_from_diff_line(&line) {
-                // TODO: cache syntaxes?
-                Some(extension) => assets.syntax_set.find_syntax_by_extension(extension),
-                None => None,
-            };
-        } else if line.starts_with("commit") {
-            state = State::Commit;
-        } else if line.starts_with("@@") {
-            state = State::DiffHunk;
-        } else if state == State::DiffHunk {
-            match syntax {
-                Some(syntax) => {
-                    paint::paint_line(line, syntax, &paint_config, &mut output);
-                    writeln!(writer, "{}", output)?;
-                    output.truncate(0);
-                    did_emit_line = true;
-                }
-                None => (),
-            }
-        }
-        if !did_emit_line {
-            writeln!(writer, "{}", raw_line)?;
-        }
-    }
-    Ok(())
-}
-
 fn process_command_line_arguments<'a>(
     assets: &'a HighlightingAssets,
     opt: &'a Opt,
@@ -206,6 +160,52 @@ fn process_command_line_arguments<'a>(
     )
 }
 
+fn delta(
+    lines: impl Iterator<Item = String>,
+    paint_config: &paint::Config,
+    assets: &HighlightingAssets,
+) -> std::io::Result<()> {
+
+    let mut syntax: Option<&SyntaxReference> = None;
+    let mut output = String::new();
+    let mut output_type =
+        OutputType::from_mode(PagingMode::QuitIfOneScreen, Some(paint_config.pager)).unwrap();
+    let writer = output_type.handle().unwrap();
+    let mut state = State::Unknown;
+    let mut did_emit_line: bool;
+
+    for raw_line in lines {
+        let line = strip_ansi_codes(&raw_line).to_string();
+        did_emit_line = false;
+        if line.starts_with("diff --") {
+            state = State::DiffMeta;
+            syntax = match parse_diff::get_file_extension_from_diff_line(&line) {
+                // TODO: cache syntaxes?
+                Some(extension) => assets.syntax_set.find_syntax_by_extension(extension),
+                None => None,
+            };
+        } else if line.starts_with("commit") {
+            state = State::Commit;
+        } else if line.starts_with("@@") {
+            state = State::DiffHunk;
+        } else if state == State::DiffHunk {
+            match syntax {
+                Some(syntax) => {
+                    paint::paint_line(line, syntax, &paint_config, &mut output);
+                    writeln!(writer, "{}", output)?;
+                    output.truncate(0);
+                    did_emit_line = true;
+                }
+                None => (),
+            }
+        }
+        if !did_emit_line {
+            writeln!(writer, "{}", raw_line)?;
+        }
+    }
+    Ok(())
+}
+
 fn compare_themes(assets: &HighlightingAssets) -> std::io::Result<()> {
     let mut opt = Opt::from_args();
     let mut input = String::new();
